### PR TITLE
Horizon Detection by Threshold

### PR DESCRIPTION
**Pull request by lamki866**
### Changes
- Wrote a new algorithm that uses a binary-threshold to segment the sky/horizon/sunrise/sunset from the foreground. Further details can be found in #96 
- Streamlined test.py to use the new algorithm, album of results: https://imgur.com/a/bbFGL
- Raw test images can be downloaded here: https://drive.google.com/file/d/1HdPBEWozHjGKDIKbQey4To7Memu7kEUD/view

### CODE REVIEW

File: aerolyzer/horizon.py [hunk 1]
```diff
@@ -1,45 +1,35 @@
...
-	for image in files:
-		img_noblur = cv2.imread(str(directory)+image)
-		img_noblur = cv2.cvtColor(img_noblur, cv2.COLOR_BGR2GRAY)
-		self.getHorizon(str(directory), img_noblur,image)
+	
+def Horizon(directory, image, output=0):
+	imgOg = cv2.imread(str(directory)+image) # Read image
+	
+	img_YUV = cv2.cvtColor(imgOg,cv2.COLOR_BGR2YCR_CB)	# Convert from BGR to YCRCB
+	b, r, g = cv2.split(img_YUV) 						# Split into blue-green-red channels
```
Comment (kwhitehall) [inline aerolyzer/horizon.py]: this line and comment is misleading. The images has been converted to YCbCr, thus, it is being split into Y, Cb and Cr, not BGR

File: aerolyzer/horizon.py [hunk 2]
```diff
@@ -1,45 +1,35 @@
...
+	imgBlueEqu = cv2.merge((cv2.equalizeHist(b), r, g))	# Equalize Blue Channel & Merge channels back
```
Comment (kwhitehall) [inline aerolyzer/horizon.py]: This variable (imgBlueEqu) is never used.

Comment (kwhitehall) [reply aerolyzer/horizon.py]: Why equalize on the Y (brightness) channel?

File: aerolyzer/horizon.py [hunk 3]
```diff
@@ -1,45 +1,35 @@
...
+	img_BGR = cv2.cvtColor(imgOg,cv2.COLOR_YCR_CB2BGR)	# Convert from YCRCB to BGR
+
+	blur = cv2.GaussianBlur(img_BGR,(5,5),0)			# Perform a gaussian blur
+
+	img_GREY = cv2.cvtColor(blur,cv2.COLOR_BGR2GRAY)	# Convert to Greyscale
+
+	ret, thresh = cv2.threshold(img_GREY,120,125,cv2.THRESH_BINARY)	# Threshold image to segment sky
```
Comment (kwhitehall) [inline aerolyzer/horizon.py]: How were the values for the threshold chosen?

Comment (lamki866) [reply aerolyzer/horizon.py]: Values for threshold were chosen through ad hoc guess and check.

File: aerolyzer/horizon.py (hunk 2 above)
Comment (lamki866) [reply aerolyzer/horizon.py]: I made a syntax mistake on line 11. 

The overall purpose of these are to normalize brightness variance between different lighting conditions in different images. Further research shows that my previous approach was unnecessary. I have now simplified lines 9-12 to use opencv's histogram equalization function.  

https://github.com/lamki866/Aerolyzer/blob/b154d392e6d99c506cbc7bb002d14fca7412585c/aerolyzer/horizon.py#L21

**Comment by lamki866** [comment]
b154d39 include changes to code in accordance with the aerolyzer pylint file. I also had to add opencv to `extension-pkg-whitelist`. 

I'm having trouble understanding the naming conventions.
For example, are local variables that are used as strings supposed to be named `stringName` or local int variables `intName'? I think the cv2.histogram() function returns a void, so I'm not sure how to name it...

File: aerolyzer/horizon.py [hunk 4]
```diff
@@ -1,35 +1,47 @@
...
+
+    _, contours, _ = cv2.findContours(dilate, cv2.RETR_LIST, cv2.CHAIN_APPROX_NONE)
+
+    contour = [(cv2.contourArea(contour), contour) for contour in contours]
+    if len(contour) > 0:
+        # Get largest contour
+        biggest = max(contour, key=lambda x: x[0])[1]
+        print "\t"+str(image)+"\t"+str(cv2.contourArea(biggest))+" \t\tPASS"
+
+        if output != 0:
```
Comment (kwhitehall) [inline aerolyzer/horizon.py]: which line is output assigned a value?

**Pull request closed without merging**